scipy/stats/stats.py

```diff
@@ -5297,12 +5329,14 @@ def kstest(rvs, cdf, args=(), N=20, alternative='two-sided', mode='approx'):
           * 'two-sided'
           * 'less': one-sided, see explanation in Notes
           * 'greater': one-sided, see explanation in Notes
-    mode : {'approx', 'asymp'}, optional
+    mode : {'auto', 'exact', 'approx', 'asymp'}, optional
         Defines the distribution used for calculating the p-value.
-        The following options are available (default is 'approx'):
+        The following options are available (default is 'auto'):
 
-          * 'approx': use approximation to exact distribution of test statistic
-          * 'asymp': use asymptotic distribution of test statistic
+          * 'auto' : uses 'exact' for small size arrays, 'asymp' for large.
+          * 'exact' : uses the exact distribution of test statistic.
+          * 'approx' : approximates the two-sided probability with twice the one-sided probability
```

Comment:
what is the use case of approx?

Reply:
`approx` used to return either the asymptotic value `kstwobign.sf(D * np.sqrt(N))`, or twice the one-sided probability `2 * ksone.sf(D, N)`, depending on whether `N > 2666` and other criteria which you asked about below.
Now `approx` always means `2 * ksone.sf(D, N)`.
`auto` will choose one of  [`exact` or `asymp`], never `approx`.
The retention of `approx` is required for backwards compatibility.
I'd expect new  code calling `kstest()` to use the default `mode=auto`.

Reply:
OK, maybe add a comment to the docstring to clarify that this option is mainly for backwards compatability and it is not needed in most situations (maybe some users will test how well the approximation works if they are familiar with the test). The approx option is not available in mstats for that reason I assume